sys/net/routing/nhdp/iib_table.c

```diff
@@ -216,6 +234,138 @@ void iib_propagate_nb_entry_change(nib_entry_t *old_entry, nib_entry_t *new_entr
+                ls_entry->dat_total[0] = 1;
+            }
+            else {
+                /* Don't add values to the queue for duplicate packets */
+                if (seq_no != ls_entry->last_seq_no) {
+                    uint16_t seq_diff;
+                    if (seq_no < ls_entry->last_seq_no) {
+                        seq_diff = (uint16_t) ((((uint32_t) seq_no) + 0xFFFF) - ls_entry->last_seq_no);
+                    }
+                    else {
+                        seq_diff = seq_no - ls_entry->last_seq_no;
+                    }
+                    ls_entry->dat_total[0] +=
+                            (seq_diff > NHDP_SEQNO_RESTART_DETECT) ? 1 : seq_diff;
+                    ls_entry->dat_received[0]++;
+                }
+            }
+
+            ls_entry->last_seq_no = seq_no;
+            ls_entry->lost_hellos = 0;
+
+            if (ls_entry->hello_interval != 0) {
+                timex_t now, i_time;
+                vtimer_now(&now);
+                i_time = timex_from_uint64(rfc5444_timetlv_decode(ls_entry->hello_interval)
+                        * MS_IN_USEC * DAT_HELLO_TIMEOUT_FACTOR);
+                ls_entry->dat_time = timex_add(now, i_time);
+            }
+
+            /* Refresh metric value for link tuple and corresponding neighbor tuple */
+            if (ls_entry->nb_elt) {
+                if ((metric_out <= ls_entry->nb_elt->metric_out) ||
+                        (ls_entry->nb_elt->metric_out == NHDP_METRIC_UNKNOWN)) {
+                    /* Better value, use it also for your neighbor */
+                    ls_entry->nb_elt->metric_out = metric_out;
+                }
+                else if (ls_entry->metric_out == ls_entry->nb_elt->metric_out){
+                    /* The corresponding neighbor tuples metric needs to be updated */
+                    iib_base_entry_t *base_elt;
+                    iib_link_set_entry_t *ls_elt;
+                    ls_entry->nb_elt->metric_out = metric_out;
+                    LL_FOREACH(iib_base_entry_head, base_elt) {
+                        LL_FOREACH(base_elt->link_set_head, ls_elt) {
+                            if ((ls_elt->nb_elt == ls_entry->nb_elt) && (ls_elt != ls_entry)) {
+                                if (ls_elt->metric_out < ls_entry->nb_elt->metric_out) {
+                                    /* Smaller DAT value is better */
+                                    ls_entry->nb_elt->metric_out = ls_elt->metric_out;
+                                }
+                                break;
+                            }
+                        }
+                    }
+                }
+            }
+            ls_entry->metric_out = metric_out;
+            break;
+#endif
+        default:
+            /* NHDP_METRIC is not set properly */
+            DEBUGF("[WARNING] Unknown NHDP_METRIC setting\n");
+            break;
+    }
+}
+
+void iib_process_metric_refresh(void)
+{
+    switch (NHDP_METRIC) {
```

Comment:
Switch case on a define is a bit pointless, isn't it? (I mean this can be done at build time or am I missing something?)